test/mtevbusted/child.lua

```diff
@@ -57,6 +57,42 @@ function TestProc:waitfor(key, timeout)
   return line
 end
 
+function TestProc:capturecommand(props)
+  if self.proc ~= nil then error("can't start already started proc") end
+  local proc, in_e, out_e, err_e =
+    mtev.spawn(self.path, self.argv, self.env)
+  self.start, self.output = mtev.uuid(), mtev.uuid()
+  self.proc = proc
+  if proc ~= nil then
+    in_e:close()
+    err_e:close()
+    mtev.coroutine_spawn(function()
+      local out_e = out_e:own()
+      local output = ''
+      mtev.waitfor(self.start,1)
+      while true do
+        local line = out_e:read("\n")
+        if line == nil then
+          break
+        end
+        output = output .. line
+      end
+      mtev.notify(self.output, output)
+      return nil
+    end)
+  else
+    error("cannot start proc")
+  end
+  mtev.notify(self.start, true)
+  local key, data = mtev.waitfor(self.output,self.timeout)
+
+  self.proc:kill()
+  self.proc:wait(10)
+  self.proc = nil
+
+  return data
+end
```

Comment:
I think this should capture the exit status (from `self.proc:wait()`), the standard output (like you have) and the stderr output. Also, `self.proc:wait(10)` - I think the `10` should be a parameter, maybe in `props`?

Reply:
self.proc:wait(10) is a direct copy/paste from the "start()" function - if we change it here, we'd probably want to change it there as well so we don't have inconsistent behavior.

As for capturing stderr - would have to look into that a bit, not sure how that'd work with the current structure to read data/notice errors.

Reply:
Should be addressed now